unit_test/blas/Test_Blas2_serial_gemv.hpp

```diff
@@ -1,388 +1,109 @@
-
-  // GEMV tolerance for scalar types
-  static double epsilon(float) { return 2 * 1e-5; }
-  static double epsilon(double) { return 1e-7; }
-  static double epsilon(int) { return 0; }
-  // tolerance for derived types
-  template <class ScalarType>
-  static double epsilon(Kokkos::complex<ScalarType>) {
-    return epsilon(ScalarType{});
-  }
-  template <class ScalarType, int VecLen>
-  static double epsilon(simd_vector<ScalarType, VecLen>) {
-    return epsilon(ScalarType{});
-  }
+            class Device, class ScalarType>
+  using functor_type =
+      SerialGEMVOp<ViewTypeA, ViewTypeX, ViewTypeY, ScalarType, AlgoTag>;
 
-  template <class ViewTypeA, class ViewTypeX, class ViewTypeY>
-  static void fill_inputs(ViewTypeA A, ViewTypeX x, ViewTypeY y) {
-    using exec_space = typename Device::execution_space;
-    Kokkos::Random_XorShift64_Pool<exec_space> rand_pool(13718);
-    fill_random_view(A, rand_pool);
-    fill_random_view(x, rand_pool);
-    fill_random_view(y, rand_pool);
-  }
+  using algorithms = std::tuple<KokkosBlas::Algo::Gemv::Unblocked,
+                                KokkosBlas::Algo::Gemv::Blocked>;
+  static bool is_supported_mode(char trans) { return true; }
 };
 
-template <class ScalarA, class ScalarX, class ScalarY, class Device,
-          class ScalarCoef = void>
-struct SerialGEMVTest {
-  static void run(const char *mode) {
-    using base =
-        SerialGEMVTestBase<ScalarA, ScalarX, ScalarY, Device, ScalarCoef>;
-    base::template run_layouts<KokkosBlas::Algo::Gemv::Unblocked>(mode);
-    base::template run_layouts<KokkosBlas::Algo::Gemv::Blocked>(mode);
-  }
-};
-
-// Special handling of Vector<SIMD<T>> (instead of plain scalars)
-// Note: MKL compact routines don't allow mixed scalar types
-template <class ScalarType, int VecLen, class Device, class ScalarCoef>
-struct SerialGEMVTest<simd_vector<ScalarType, VecLen>,
-                      simd_vector<ScalarType, VecLen>,
-                      simd_vector<ScalarType, VecLen>, Device, ScalarCoef> {
-  static void run(const char *mode) {
-    using vector_type = simd_vector<ScalarType, VecLen>;
-    using base = SerialGEMVTestBase<vector_type, vector_type, vector_type,
-                                    Device, ScalarCoef>;
-    // run all usual, non-vector tests
-    base::template run_layouts<KokkosBlas::Algo::Gemv::Unblocked>(mode);
-    base::template run_layouts<KokkosBlas::Algo::Gemv::Blocked>(mode);
-    // run vector tests
 #ifdef __KOKKOSBLAS_ENABLE_INTEL_MKL_COMPACT__
-    base::template run_layouts<KokkosBlas::Algo::Gemv::CompactMKL>(mode);
-#endif
-  }
+struct SerialMKLGemvFactory {
+  template <class AlgoTag, class ViewTypeA, class ViewTypeX, class ViewTypeY,
+            class Device, class ScalarType>
+  using functor_type =
+      SerialGEMVOp<ViewTypeA, ViewTypeX, ViewTypeY, ScalarType, AlgoTag>;
+
+  using algorithms = std::tuple<KokkosBlas::Algo::Gemv::CompactMKL>;
+  static bool is_supported_mode(char trans) { return trans != 'X'; }
```

Comment:
@lucbv Let me propose simpler, non-templated mode filter that blocks `ConjNoTranspose` on MKL.
PS. `SerialMKLGemvFactory` is great!